File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
import sys
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        buy = sys.maxsize
        for i in range(0, len(prices)):
            buy = min(buy, prices[i])
            profit = max(profit, prices[i] - buy)
        return profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int result = 0;
        int buy = Integer.MAX_VALUE;
        for(int i=0;i<prices.length;i++){
            buy = Math.min(buy, prices[i]);
            result = Math.max(prices[i] - buy, result);
        }
        return result;
    }
}